.github/workflows/deploy.yml

```diff
@@ -0,0 +1,276 @@
+# Do not rename this file. The name "deploy.yml" is known to
+# npm for trusted OIDC publishing.
+name: Deploy
+
+# This workflow has safeguards to prevent exploitation.
+# All non-blocked jobs MUST check that the workflow_run event is
+# 'push' (i.e., github.event.workflow_run.event == 'push') to prevent
+# forked PRs from modifying enough to trigger a deployment by force.
+# Forked PRs will only run with 'pull_request' events within our runs.
+on: # zizmor: ignore[dangerous-triggers]
+  workflow_run:
+    workflows:
+      - Tests
+    types:
+      - completed
+    branches:
+      - master
+      - develop
+
+concurrency:
+  group: deploy/${{ github.ref_name }}
+  cancel-in-progress: false
+
+permissions:
+  contents: read
+
+jobs:
+  deploy-next:
+    name: Deploy "next" to npm
+    if: ${{ github.event.workflow_run.conclusion == 'success' && github.event.workflow_run.head_branch == 'develop' && github.event.workflow_run.event == 'push' }}
+    environment:
+      name: registry.npmjs.org
+    permissions:
+      contents: read
+      id-token: write # Required for OIDC
+    runs-on: ubuntu-24.04
+    outputs:
+      version: ${{ steps.determine-version.outputs.version }}
+      packageName: ${{ steps.determine-version.outputs.name }}
+    steps:
+      - &checkout
+        name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          persist-credentials: false
+      - &install-deps
+        name: Install Dependencies
+        uses: ./.github/actions/install-deps
+      - &build
+        name: Build
+        run: |
+          npm run prepare
+          npm run build
+      - name: Determine prerelease version
+        id: determine-version
+        run: |
+          echo "::group::Determining new prerelease version"
+          NAME=$(npm pkg get name | tr -d '"')
+          LATEST_VERSION=$(npm pkg get version | tr -d '"')
+          echo "Latest version in package.json: $LATEST_VERSION"
+
+          # add 8 characters from the current SHA as prerelease identifier
```

Comment:
The circleci version used 7 chars, not 8. We shouldn't change the format without a good reason

Reply:
It was 0 to 7, meaning 8 characters. I'll double check what is on NPM once the Cloudflare outage is resolved and if it was pulling 7 to submit then it's an easy update to make.